### ISSUE TITLE
Check boxes & radio button

### ISSUE THREAD

**Issue by SarahHornsey** [Financial-Times/o-forms#69]
Hello,

We would like to make changes to the check boxes and radio buttons. 

I've attached a design....
![org007-radio-and-check-v2](https://cloud.githubusercontent.com/assets/8235225/5107243/e44a63ca-6ff7-11e4-89fa-33fe4a79b8a7.png)
Any questions let me know....

Thanks,
Sarah

**Comment by kaelig** [comment]
The interactive graphics team needs checkboxes that don't "pop" too much, but the green color of these new checkboxes carries a lot of meaning and draws a lot of attention, when what they want is to put the content first.

We'll need to have the two styles in the forms module:
- the new style with green active states, great for standalone forms such as the registration form
- the old style, simpler with a black tick and white background, great for UI elements that shouldn't steal the user's attention from the content

What should the default style be?

@SarahHornsey @danskinner @kavanagh what are your thoughts on this?

**Comment by SarahHornsey** [comment]
Hi Kaelig,

Yes - I agree regards the check boxes for interactive graphics as the green will stand out far too much. Especially on long lists etc...

@danskinner - what are your thoughts?

**Comment by kavanagh** [comment]
What about a container class, something like `o-forms--in-editorial`? All buttons and inputs inside this container have pop-reduction.

**Comment by kaelig** [comment]
@kavanagh Parent modifiers add overhead for applying styles at a more granular level. I suggest a modifier directly on the checkboxes / radio buttons themselves.

**Comment by kavanagh** [comment]
Should buttons be part of forms?

**Comment by kaelig** [comment]
Sorry by "radio buttons" I meant "radio controls"

**Comment by kavanagh** [comment]
I know, I was just wondering if there is so much shared between o-buttons and o-forms then maybe bring them together into one component.

**Comment by kaelig** [comment]
That's true there are a lot of similarities. Feel free to open an issue on the subject, as this thread is about radio controls.

**Comment by danskinner** [comment]
I think the default should be the green with a no-colour option.

**Comment by kaelig** [comment]
@danskinner I should have mentioned this:

In CSS it's (much) easier to make it neutral and then make a modifier that adds the green, rather than styling them up and then "resetting" styles to make them neutral.

Also it makes upgrading from the v0.x.x to the v1.0.0 easier.

Would that be okay with you?
```html
<!-- Default neutral checkboxes: -->
<input type="radio" class="o-forms-radio" />
<input type="checkbox" class="o-forms-checkbox" />

<!-- "pop" mode: Make the checkbox or radio control look more prominent when checked: -->
<input type="radio" class="o-forms-radio o-forms-radio--pop" />
<input type="checkbox" class="o-forms-checkbox o-forms-checkbox--pop" />
```

Now "pop" isn't a great name, so if you agree with the principle you'll also have to bring a name :)

**Comment by kaelig** [comment]
Please browse the demos with two styles (green = "highlighted"):
- [Checkboxes](http://build.origami.ft.com/files/o-forms@1.0.0-beta.9/demos/checkboxes.html)
- [Radio controls](http://build.origami.ft.com/files/o-forms@1.0.0-beta.9/demos/radios.html)

If there is a strong case for having green as a default style, and that it's important for the general UX despite the bad impact on code, let me know.

**Comment by danskinner** [comment]
Does it not makes sense to use the same naming conventions as with the buttons?  Tthere the default is coloured and then there's an '-uncolored' variant with colour removed for IG.

**Comment by kaelig** [comment]
I see what you mean. That's a good point.

In the case of buttons we have a pretty simple border and color override:

### Default
![screenshot 2015-03-17 16 59 01](https://cloud.githubusercontent.com/assets/85783/6692822/01646d8a-ccc7-11e4-8e78-cf7bfbc97405.png)

### Uncoloured
![screenshot 2015-03-17 16 59 16](https://cloud.githubusercontent.com/assets/85783/6692826/04a3231a-ccc7-11e4-884e-8f36df9f78a0.png)

So it makes sense in that case.

----

Usually the default style is the most common use: if we think there's going to be more usage with the green checkboxes, then it should be the default state. We have to remember that most of our websites are content-centric, in which forms are not at the center of the user's attention. The uncoloured checkboxes and radio buttons seem to be the subtler choice to avoid stealing the focus from the content.

Basically:
1. It makes more sense from a CSS perspective
2. I'm afraid we're going to see checkboxes that are very prominent steal the show from the content because developers wouldn't look twice at what options they have to build their forms with

PS: the naming convention can still change, as I'm not sure "highlight" is a great name anyway

**Comment by kaelig** [comment]
Discussed this morning: we'll keep the uncoloured checkboxes as default and the green ones are usable on an opt-in basis.
